### PR TITLE
Halloween event map

### PR DESCRIPTION
**Pull request by Eyeveri**
Two large towers on Xanu. WIP. Having an extremely hard time with getting spawning to work. Help welcome.

### CODE REVIEW

**Comment by Eyeveri** [comment]
!wip

File: maps/event/halloween/halloweenareas.dm
Comment (SleepyGemmy) [inline maps/event/halloween/halloweenareas.dm:1]: change file name from `halloweenareas` to `halloween_areas`.

File: aurorastation.dme [hunk 1]
```diff
@@ -3634,6 +3634,8 @@
 #include "maps\event\code\event.dm"
 #include "maps\event\generic_dock\code\generic_dock.dm"
 #include "maps\event\generic_dock\code\generic_dock_areas.dm"
+#include "maps\event\halloween\halloween.dm"
+#include "maps\event\halloween\halloweenareas.dm"
```
Comment (SleepyGemmy) [inline aurorastation.dme:3638]:
```suggestion
#include "maps\event\halloween\halloween_areas.dm"
```

File: code/game/ambience.dm [hunk 2]
```diff
@@ -155,6 +155,13 @@
 	'sound/ambience/expoutpost/expoutpost4.ogg'\
 	)
 
+//Defines Tension Ambience
+#define AMBIENCE_TENSION list(\
+    'sound/ambience/tension/tension.ogg',\
+    'sound/ambience/tension/argitoth.ogg',\
+    'sound/ambience/tension/horror.ogg',\
+    'sound/ambience/tension/burning_terror.ogg'\
+    )
```
Comment (SleepyGemmy) [inline code/game/ambience.dm:164]:
```suggestion
//Defines Tension Ambience
#define AMBIENCE_TENSION list(\
    'sound/ambience/tension/tension.ogg',\
    'sound/ambience/tension/argitoth.ogg',\
    'sound/ambience/tension/horror.ogg',\
    'sound/ambience/tension/burning_terror.ogg'\
    )

```

File: maps/event/halloween/halloweenareas.dm
Comment (SleepyGemmy) [inline maps/event/halloween/halloweenareas.dm:1]: this file should use boolean defines where applicable (replace 1 with TRUE and 0 with FALSE).

**Review by SleepyGemmy — COMMENTED** [review]
**some general comments**
theres some props with proper names although they should be improper, as well as some with prefixes and punctuation (e.g. `An ugly blob!`).
proper names dont matter too much but prefix and punctuation is and should be handled by code.

**Comment by Greenjoe12345** [comment]
image
image
image
various stairs need to be altered to use the new stair icons

**Comment by Eyeveri** [comment]
Okay, I'm going to mark this as a review so the map can be officially added to code. I will get a changelog up and stuff. 

Any feedback is welcome to make it more improved, too.

**Comment by Eyeveri** [comment]
!review

File: code/controllers/subsystems/ticker.dm [hunk 3]
```diff
@@ -453,7 +453,7 @@ var/datum/controller/subsystem/ticker/SSticker
 			available_ghostroles |= G.name
 
 	// Special case, to list the Merchant in case it is available at roundstart
-	if(SSjobs.type_occupations[/datum/job/merchant].total_positions)
+	if(SSjobs.type_occupations[/datum/job/merchant]?.total_positions)
```
Comment (SleepyGemmy) [inline code/controllers/subsystems/ticker.dm:456]: why is this being changed?

Comment (Geevies) [reply code/controllers/subsystems/ticker.dm:456]: this causes a runtime when the merchant job isn't one of the available jobs on, say, an event map

Comment (SleepyGemmy) [reply code/controllers/subsystems/ticker.dm:456]: thanks for the explanation!

**Review by SleepyGemmy — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**